server/server.go

```diff
@@ -769,13 +833,14 @@ func (s *Server) copyInfo() Info {
 
 func (s *Server) createClient(conn net.Conn) *client {
 	// Snapshot server options.
+	// TODO(dlc) - This can get expensive.
```

Comment:
No it is not. There is no copy/clone. s.getOpts() is just returning the pointer to the options under the optsMu lock, so that it is safe with the reload code which switch s.opts with new options pointer under that same lock.

Reply:
Ah ok, much better then ;)